frontend/src/components/mainPage/Done.tsx

```diff
@@ -1,17 +1,23 @@
 import { Box, Stack, Typography } from "@mui/joy";
 import { useCurrentUser } from "../../lib/useCurrentUser";
-import { TaskBox } from "../TaskBox";
-// import { useEffect, useState } from "react";
-// import { User } from "../../../../api";
-// import { config } from "../../config";
-// import { useFetch } from "../../lib/useFetch";
-
-// TODO Task 2 - implementa la chiamata api per recuperare il destinatario del regalo
+import { useEffect, useState } from "react";
+import { useFetch } from "../../lib/useFetch";
 
 export const Done: React.FC = () => {
   const currentUser = useCurrentUser();
+  const fetch = useFetch();
+
+  const [recipient, setRecipient] = useState<{ first_name: string; last_name: string } | null>(null);
 
-  // const fetch = useFetch();
+  useEffect(() => {
+    if (currentUser?._id) {
+      fetch(`/api/recipient/${currentUser._id}`).then((response: any) => {
```

Comment:
Tutto ok, se non che questa api non l'hai implementata!
L'api da implementare era /api/recipient, senza parametri, in quanto l0utente corrente è già in sessione.